opencog/atoms/core/Variables.cc

```diff
@@ -737,6 +737,28 @@ void Variables::get_vartype(const Handle& htypelink)
 				Type vt = TypeNodeCast(ht)->get_kind();
 				if (ATOM != vt) typeset.insert(vt);
 			}
+			else if (TYPE_INH_NODE == var_type)
+			{
+				Type vt = TypeNodeCast(ht)->get_kind();
+				if (ATOM != vt)
+				{
+					TypeSet ts;
+					auto i_it = std::inserter(ts, ts.begin());
+					nameserver().getChildrenRecursive(vt, i_it);
+					typeset.insert(ts.begin(), ts.end());
+				}
+			}
+			else if (TYPE_CO_INH_NODE == var_type)
+			{
+				Type vt = TypeNodeCast(ht)->get_kind();
+				if (ATOM != vt)
```

Comment:
there should also be an `else` statement: else if vt is Atom, then bottom type, i.e. the match will always fail.

Reply:
Never mind ... handling this correctly is actually hard to do. There is a very special case, here: `(TypeChoice (TypCoInh 'Atom))`